# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/TextDirectiveUtil.h
# Commit: a9560be525c8
# Full Hash: a9560be525c8880189debb175e8ad3fa9e56ba49
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-09-10 21:28:29
# Regressor Bug: 1979588
# File Overlap Count: 2
# Overlapping Files: dom/base/test/test_text-fragments-create-text-directive.html, dom/base/TextDirectiveCreator.cpp
# Description:
#   Bug 1979588 - Text Fragments: Correctly deal with long words at end of range for range-based matching. r=dom-core,smaug
#   
#   The range-based approach for creating a text directive uses the word breaker algorithm
#   to identify where to split between `start` and `end` term. If the range ends with a word
#   which is over 50% of the length content, the algorithm would return wordEnd=length,
# ==============================================================================

diff -r b6966486a1dd -r a9560be525c8 dom/base/TextDirectiveUtil.h
--- a/dom/base/TextDirectiveUtil.h	Wed Sep 10 12:06:33 2025 +0000
+++ b/dom/base/TextDirectiveUtil.h	Wed Sep 10 12:33:30 2025 +0000
@@ -551,8 +551,10 @@
     return 0;
   }
 
-  MOZ_ASSERT(!nsContentUtils::IsHTMLWhitespace(aReferenceString.First()));
-  MOZ_ASSERT(!nsContentUtils::IsHTMLWhitespace(aReferenceString.Last()));
+  MOZ_DIAGNOSTIC_ASSERT(
+      !nsContentUtils::IsHTMLWhitespace(aReferenceString.First()));
+  MOZ_DIAGNOSTIC_ASSERT(
+      !nsContentUtils::IsHTMLWhitespace(aReferenceString.Last()));
   uint32_t referenceStringPosition =
       direction == TextScanDirection::Left ? aReferenceString.Length() - 1 : 0;
 